doc: Download Materialize Emulator (#29043)
MaterializeInc/materialize: doc/user/content/get-started/install-materialize.md

---

> title: "Install Materialize locally"

Reviewer: ```suggestion
title: "Run Materialize on Docker"
```

> description: "Download the Docker image and run Materialize locally for evaluation"

Reviewer: ```suggestion
description: "Run a containerized version of Materialize for evaluation or testing using Docker."
```

menu:
  main:
    parent: "get-started"
    weight: 16
    name: "Install Materialize locally"

---

For testing/evaluation purposes, you can download the Materialize Docker image

> and run Materilize locally in a Docker container.  The Docker image:

Author: When the table is ready, it'll actually come here at the beginning, and that will cover an earlier callout of this distinction.

> and run Materilize locally in a Docker container.  The Docker image:

Author: Oh ... and the "locally" was to be inline with what we're going with on our site of "Try Materialize locally" but am good either way, and while at it, fixed my misspelling of Materialize.

- Includes the [Quickstart](/get-started/quickstart) data source

- Supports connecting your data sources

- Provides [community support via

>   Slack](https://materializecommunity.slack.com/join/shared_invite/zt-2bad5ce4i-ZsiPWI5jd7Q9pRDGYj3dkw#/shared-invite/email)

Reviewer: Should we include the comparison table @ggnall created somewhere as an overview here, just to make it clearer? Then we can elaborate more each of the characteristics under `Limitations`.

Author: Yup ... but, until the text in the table is final, figured I won't include it until then.

> To more fully evaluate
> Materialize and its features, Materialize offers a [free trial
> account](https://materialize.com/register/?utm_campaign=General&utm_source=documentation).

Reviewer: ```suggestion
The Docker image is not representative of Materialize's performance and full feature set. To evaluate
Materialize for production scenarios, we recommend signing up for a [free trial
account](https://materialize.com/register/?utm_campaign=General&utm_source=documentation).
```

{{< warning >}}

The Docker image is for <redb> testing/evaluation purposes only. It is not
suitable for production deployments</redb>.  See also [Limitations](#limitations).

{{</ warning >}}

> ### License and Privacy Policy

Reviewer: ```suggestion
### License and privacy policy
```

- Use of the Docker image is subject to Materialize's [BSL
  License](https://github.com/MaterializeInc/materialize/blob/main/LICENSE).

- By downloading the Docker image, you are agreeing to Materialize's [privacy
  policy](https://materialize.com/privacy-policy/).

### Prerequisites

- Docker. If [Docker](https://www.docker.com/) is not installed, refer to its
[official documentation](https://docs.docker.com/get-docker/) to install.

### Run Materialize in a Docker container

{{< note >}}

- Use of the Docker image is subject to Materialize's [BSL
   License](https://github.com/MaterializeInc/materialize/blob/main/LICENSE).

- By downloading the Docker image, you are agreeing to Materialize's [privacy
   policy](https://materialize.com/privacy-policy/).

> {{</ note >}}

Reviewer: Do we need to include this also as a separate section above, if we're already including it here?

Author: I figured for those people who will skip that above section, having it repeated right where they will be performing the steps could catch some people.

1. In a terminal, issue the following command to run a Docker container from the
   Materialize image. The command downloads the Materialize image, if one has
   not been already downloaded.

   ```sh
   docker run -v mzdata:/mzdata -p 6875:6875 -p 6876:6876 materialize/materialized
   ```

   When running locally:

   - Materialize logs are emitted to the `stderr`.
   - The SQL interface is available on port `6875`.

>    - The HTTP interface is available on port `6876`.

Reviewer: Do we need to mention the HTTP interface, since it's full-on legacy? I think it's potentially confusing + we should not give users the false impression that this is anything remotely close to the information displayed in the Materialize Console. Also, we for sure don't want to be fielding questions about it.

Author: Will remove. cc @ggnall  in case he wants to remove it from the downloads page as well.

   - A default user `materialize` is created.
   - A default database `materialize` is created.

1. Connect to Materialize using your preferred SQL client, using the following
   connection field values:

   | Field    | Value         |
   |----------|---------------|
   | Server   | `localhost`   |
   | Database | `materialize` |
   | Port     | `6875`        |
   | Username | `materialize` |

>    For example, if using `psql`

Reviewer: ```suggestion
   For example, if using `psql`:
```

Reviewer: @ggnall, reminder for the future that we should push users to `mz`. Also, @kay-kim: we used to provide instructions to install `psql`, because most users did not have it installed. You can point people to the [SQL clients page](https://materialize.com/docs/integrations/sql-clients/#psql).

<img width="827" alt="Screenshot 2024-08-19 at 17 48 30" src="https://github.com/user-attachments/assets/9a5b87a4-0ba1-4696-b46c-cf783b78622b">

Author: 👍  I'll link to integrations/sql-clients/#psql. 
Since people are downloading for their eval, I'm hoping (recognizing hope and reality are 2 different things) they'll connect with the client they have on hand and just use the example statement really as just one example as opposed to a suggestion.  (but, definitely, doubly-recognizing that the reality in practice may differ).

   ```sh
   psql postgres://materialize@localhost:6875/materialize
   ```

1. Once connected, you can get started with the
   [Quickstart](/get-started/quickstart). Where a command for the Materialize
   console can differ from other SQL clients, both versions of the command are
   presented.

### Next steps

To get started ingesting your own data from an external system like Kafka, MySQL

> or PostgreSQL, check the documentation for [sources](/sql/create-source/).

Reviewer: @ggnall, re-stating that we should provided Docker Compose files that come pre-baked with these services, at some point. Otherwise, we'll spend 99.9% of our support time debugging Docker networking issues.

Reviewer 2: good call. noting!

To learn more about Materialize, sign up for a [free
trial](https://materialize.com/register/?utm_campaign=General&utm_source=documentation)
or [schedule a
demo](https://materialize.com/demo/?utm_campaign=General&utm_source=documentation).

### Limitations

The Docker image is for <redb> testing/evaluation purposes only</redb>.  That is:

- It is <redb>not</redb> suitable for production deployments.

- Materialize does <redb>not</redb> provide support for the Docker image.

> Additionally, the Docker image:

Reviewer: We should provide more details about each limitation, otherwise it isn't straightforward to understand the actual impact of these limitations if you want to build something ([context](https://www.notion.so/materialize/Improving-the-get-started-experience-193f51de3ca546fe9d8dfe1d30e066b2?pvs=4#5d64a509c4d344aeb3130d320e675db8)).

### Run Materialize in a Docker container

1. Go to the

>    [Download](https://materialize.com/download?utm_campaign=General&utm_source=documentation)

Reviewer 2: I removed the redirect in https://github.com/MaterializeInc/www/pull/1183

Author: 👍 And I put back the `docker run` statement instead of going to the Download page.  As an FYI -- this page won't go live until after the /downloads page goes live.
